superset/assets/javascripts/explorev2/reducers/exploreReducer.js

```diff
@@ -90,6 +87,7 @@ export const exploreReducer = function (state, action) {
           chartUpdateEndTime: null,
           chartUpdateStartTime: now(),
           triggerQuery: false,
+          queryRequest: action.queryRequest,
```

Comment:
ok so it sounds like it is not ideal to store functions or promises in the state... http://redux.js.org/docs/faq/OrganizingState.html#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state, however, if this is working as expected, i'm not against it. i'm trying to think of another way to handle this.